providers/session_state_test.go

```diff
@@ -25,7 +25,7 @@ func TestSessionStateSerialization(t *testing.T) {
 	}
 	encoded, err := s.EncodeSessionState(c)
 	assert.Equal(t, nil, err)
-	assert.Equal(t, 3, strings.Count(encoded, ":"))
+	assert.Equal(t, 4, strings.Count(encoded, ":"))
```

Comment:
5 items in `encoded` string, 4 `:` delimiters. Looks fine to me.
The only thing that is missed here is `PermittedGroups` but they are stored in `AzureProvider` structure and shouldn't be visible in `SessionState` so we're good